Panic (SIGSEGV) at ingress inspector.

<!--

Welcome to ingress-nginx!  For a smooth issue process, try to answer the following questions.
Don't worry if they're not all applicable; just try to include what you can :-).
More info helps better understanding of the issue (needless to say).

If you need to include code snippets or logs, please put them in fenced code
blocks.  If they're super-long, please use the details tag like
<details><summary>super-long log</summary> lots of stuff </details>

-->

<!--

IMPORTANT!!!

Please complete the next sections or the issue will be closed.
This questions are the first thing we need to know to understand the context.

-->

**What happened**:

Panic in inspector =_

**What you expected to happen**:

Error message instead UB


**NGINX Ingress controller version** 

Latest code from master node.



**Reproduce**

Append testcase at `k8s.io/ingress-nginx/internal/ingress/inspector` with one more InvalidIngress:

```go
{	
	Path: "notvalidpathname-panics",
},
```
Run tests and get:
```console
Running tool: /usr/local/go/bin/go test -timeout 30s -run ^TestValidatePathType$ k8s.io/ingress-nginx/internal/ingress/inspector

--- FAIL: TestValidatePathType (0.00s)
    --- FAIL: TestValidatePathType/invalid_should_return_multiple_errors (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7c39bd]

goroutine 12 [running]:
testing.tRunner.func1.2({0x886d60, 0xcf7b60})
	/home/wooffie/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.5.linux-amd64/src/testing/testing.go:1734 +0x21c
testing.tRunner.func1()
	/home/wooffie/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.5.linux-amd64/src/testing/testing.go:1737 +0x35e
panic({0x886d60?, 0xcf7b60?})
	/home/wooffie/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.5.linux-amd64/src/runtime/panic.go:792 +0x132
k8s.io/ingress-nginx/internal/ingress/inspector.ValidatePathType(0xf?)
	/home/wooffie/work/github/ingress-nginx/internal/ingress/inspector/inspector.go:58 +0x1fd
k8s.io/ingress-nginx/internal/ingress/inspector.TestValidatePathType.func1(0xc00018da40)
	/home/wooffie/work/github/ingress-nginx/internal/ingress/inspector/inspector_test.go:185 +0x28
testing.tRunner(0xc00018da40, 0xc0000a7cc0)
	/home/wooffie/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.5.linux-amd64/src/testing/testing.go:1792 +0xf4
created by testing.(*T).Run in goroutine 7
	/home/wooffie/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.5.linux-amd64/src/testing/testing.go:1851 +0x413
FAIL	k8s.io/ingress-nginx/internal/ingress/inspector	0.006s
FAIL
```

--- a/internal/ingress/inspector/inspector.go
+++ b/internal/ingress/inspector/inspector.go
@@ -55,9 +55,11 @@ func ValidatePathType(ing *networking.Ingress) error {
 				}
 				if path.PathType == nil || *path.PathType != implSpecific {
 					if isValid := validPathType.MatchString(path.Path); !isValid {
-						pathTypeStr := "nil"
+						var pathTypeStr string
 						if path.PathType != nil {
 							pathTypeStr = string(*path.PathType)
+						} else {
+							pathTypeStr = "<nil>"
 						}
 						err = errors.Join(err, fmt.Errorf("path %s cannot be used with pathType %s", path.Path, pathTypeStr))
 					}

--- a/internal/ingress/inspector/inspector.go
+++ b/internal/ingress/inspector/inspector.go
@@ -55,7 +55,11 @@ func ValidatePathType(ing *networking.Ingress) error {
 				}
 				if path.PathType == nil || *path.PathType != implSpecific {
 					if isValid := validPathType.MatchString(path.Path); !isValid {
-						err = errors.Join(err, fmt.Errorf("path %s cannot be used with pathType %s", path.Path, string(*path.PathType)))
+						pathTypeStr := "nil"
+						if path.PathType != nil {
+							pathTypeStr = string(*path.PathType)
+						}
+						err = errors.Join(err, fmt.Errorf("path %s cannot be used with pathType %s", path.Path, pathTypeStr))
 					}
 				}
 			}

--- a/internal/ingress/inspector/inspector_test.go
+++ b/internal/ingress/inspector/inspector_test.go
@@ -101,6 +101,9 @@ var (
 									PathType: &prefix,
 									Path:     "/lala/xp\ntest",
 								},
+								{
+									Path: "notvalidpathname-panics",
+								},
 							},
 						},
 					},
@@ -174,6 +177,7 @@ func TestValidatePathType(t *testing.T) {
 				aErr("xpto/lala", "Exact"),
 				aErr("/foo/bar/[a-z]{3}", "Prefix"),
 				aErr("/lala/xp\ntest", "Prefix"),
+				aErr("notvalidpathname-panics", "nil"),
 			),
 		},
 	}

--- a/internal/ingress/inspector/inspector_test.go
+++ b/internal/ingress/inspector/inspector_test.go
@@ -177,7 +177,7 @@ func TestValidatePathType(t *testing.T) {
 				aErr("xpto/lala", "Exact"),
 				aErr("/foo/bar/[a-z]{3}", "Prefix"),
 				aErr("/lala/xp\ntest", "Prefix"),
-				aErr("notvalidpathname-panics", "nil"),
+				aErr("notvalidpathname-panics", "<nil>"),
 			),
 		},
 	}

